pkg/deploy/migration/on-reconcile-one-time-migration.go

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package migration
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/eclipse-che/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	gitSelfSignedCertsConfigMapName = "che-git-self-signed-cert"
+)
+
+type Migrator struct {
+	deploy.Reconcilable
+
+	migrationDone bool
+}
+
+func NewMigrator() *Migrator {
+	return &Migrator{
+		migrationDone: false,
+	}
+}
+
+func (m *Migrator) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if m.migrationDone {
+		return reconcile.Result{}, true, nil
+	}
+
+	result, done, err := m.migrate(ctx)
+	m.migrationDone = done && err == nil
+	return result, done, err
+}
+
+func (m *Migrator) Finalize(ctx *deploy.DeployContext) error {
+	return nil
+}
+
+func (m *Migrator) migrate(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if err := addRequiredLabelsForConfigMaps(ctx); err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	if err := addRequiredLabelsForSecrets(ctx); err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	if err := addRequiredLabelsForPartOfCheObjects(ctx); err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	// Give some time for the migration resources to be flushed
+	return reconcile.Result{RequeueAfter: 5 * time.Second}, true, nil
```

Comment:
```suggestion
	return reconcile.Result{RequeueAfter: 5 * time.Second}, false, nil
```

Reply:
We cannot do that as the migration is one-time procedure and we have to mark it as done when its end reached.

Reply:
if done == true then result is not taken into account. 
https://github.com/eclipse-che/che-operator/blob/main/controllers/che/checluster_controller.go#L274-L276

Reply:
But I see that it will ignore the `RequeueAfter`, so will fix it in the reconciler.